### PR TITLE
feat: ✨ prompt user for answer to ice breaker question

### LINKED ISSUE #31: Allow Catniss to answer a prompt
Catniss should be able to answer and submit the prompt, whether cat or catfish so that other players can see her answer and guess weather she is a catfish or a cat.

all of the cats should get the same prompt
the catfish should not receive the same prompt as the cat

- [x] read prompts from files
- [x] put prompts in firestore when round starts
- [x] display prompt to user
- [x] allow user to answer prompt
- [x] store answer in firestore
- [ ] connect to assign role

**Acceptance Criteria**
- If the user is a cat, they should be given a prompt that is shared amongst the other cats
- If the user is a catfish, they should be given a prompt different from the others, but in the same category
- Once given a prompt, they should be able to type their answer into an input field which keeps their response for the next phase
- If the user leaves the response blank or only enters a space, they should not be able to enter that as their response

### LINKED ISSUE #83: Material Components and CSS Style
Catniss likes dark mode and colorful buttons. Make the app look like a nice modern app using Google's material guidelines. This is because catniss wants to have a pleasant viewing experience 

Tasks:
- [x] get material components imported
- [x] use material for buttons
- [x] use material for text fields
- [x] spend at least 16 hours on styles (including buttons and text fields)

Acceptance Criteria

- The website should have a consistent design/style

### PR DESCRIPTION
**Pull request by nstringham**
this PR adds the code to prompt the user for an answer to an ice breaker question.

### CODE REVIEW

File: client/src/components/prompt.svelte [hunk 1]
```diff
@@ -0,0 +1,90 @@
+<script lang="ts">
```
Comment (gchamb) [inline client/src/components/prompt.svelte]: Uppercase component name

File: client/src/components/prompt.svelte [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    if (error != undefined) {
+      return;
+    }
+
+    setDoc(anserDoc, { answer });
+  }
+</script>
+
+<form class="wraper" on:submit|preventDefault={submitAnswer}>
+  <lable class="question" for="prompt-answer">{prompt}</lable>
```
Comment (gchamb) [inline client/src/components/prompt.svelte]: label?

File: client/src/components/prompt.svelte [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+  <div class="input">
+    <input id="prompt-answer" type="text" bind:value={answer} />
+    <lable class="error" for="prompt-answer">{error ?? ""}</lable>
```
Comment (gchamb) [inline client/src/components/prompt.svelte]: label?

File: client/src/components/prompt.svelte (hunk 2 above)
Comment (nstringham) [reply client/src/components/prompt.svelte]: thanks

File: firestore.rules [hunk 4]
```diff
@@ -1,9 +1,24 @@
...
+
+      match /privatePlayers/{uid} {
+        allow get: if request.auth.uid == uid;
+      }
+
+      match /promptAnswers/{uid} {
+        function validAnswer() {
+          return request.resource.data.answer is string
+            && 0 < request.resource.data.answer.size() && request.resource.data.answer.size() <= 50
+            && request.resource.data.answer == request.resource.data.answer.trim();
```
Comment (gchamb) [inline firestore.rules:17]: shouldn't you trim before you check the size of the string? Example: '   ' is technically length 3 but once trimmed is 0

Comment (gchamb) [reply firestore.rules:17]: i had the 3 spaces in between the '' github trimmed automatically

Comment (nstringham) [reply firestore.rules:17]: I am checking that the answer is equal to the trimmed answer so it doesn't really matter

File: client/src/routes/game/+page.svelte [hunk 5]
```diff
@@ -54,15 +79,19 @@
...
-    {#if lobbyData.state === "WAIT"}
-      <LobbyComponent {code} players={lobbyData.players} />
-    {:else if lobbyData.state === "PROMPT"}
-      <p>PROPMPT PAGE</p>
-    {/if}
-  </div>
-{/if}
+<!-- We could have a loading animation until the lobby is not undefined -->
+<div>
+  {#if $user == undefined || lobby == undefined || lobbyCode == null}
```
Comment (gchamb) [inline client/src/routes/game/+page.svelte]: should be == null

**Review by njumaah — CHANGES_REQUESTED** [review]
informed nate in person on what to update

**Review by njumaah — APPROVED** [review]

**Review by njumaah — DISMISSED** [review]

**Comment by njumaah** [comment]
lgtm

**Review by njumaah — APPROVED** [review]

**Pull request merged**